sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala

```diff
@@ -271,32 +272,41 @@ object JavaTypeInference {
 
       case c if listType.isAssignableFrom(typeToken) =>
         val et = elementType(typeToken)
-        MapObjects(
+        UnresolvedMapObjects(
```

Comment:
can we exclude other changes except this one? This one is very easy to reason about. We did the same thing in `ScalaReflection.`

We need more time to think about the map case, and fix it in `ScalaReflection` as well.

Reply:
Sure. Should I create another pull request with this change only?

Reply:
yes please, thanks!

Reply:
Removed other changes from this PR and created a new one with only map case.
As far as I see, everything works fine with scala classes, because StructTypes are generated based on constructor parameters, and they are available in correct order with correct names. Which is hardly achievable with Java beans..